meta/autoid/autoid.go

```diff
@@ -428,17 +456,22 @@ func NewSequenceAllocator(store kv.Storage, dbID int64, info *model.SequenceInfo
 func NewAllocatorsFromTblInfo(store kv.Storage, schemaID int64, tblInfo *model.TableInfo) Allocators {
 	var allocs []Allocator
 	dbID := tblInfo.GetDBID(schemaID)
+	idCacheOpt := CustomAutoIncCacheOption(tblInfo.AutoIdCache)
+
 	hasRowID := !tblInfo.PKIsHandle && !tblInfo.IsCommonHandle
+	if hasRowID {
+		alloc := NewAllocator(store, dbID, false, RowIDAllocType, idCacheOpt)
+		allocs = append(allocs, alloc)
+	}
 	hasAutoIncID := tblInfo.GetAutoIncrementColInfo() != nil
-	if hasRowID || hasAutoIncID {
-		if tblInfo.AutoIdCache > 0 {
-			allocs = append(allocs, NewAllocator(store, dbID, tblInfo.IsAutoIncColUnsigned(), RowIDAllocType, CustomAutoIncCacheOption(tblInfo.AutoIdCache)))
-		} else {
-			allocs = append(allocs, NewAllocator(store, dbID, tblInfo.IsAutoIncColUnsigned(), RowIDAllocType))
-		}
+	if hasAutoIncID {
+		alloc := NewAllocator(store, dbID, tblInfo.IsAutoIncColUnsigned(), RowIDAllocType, idCacheOpt)
+		allocs = append(allocs, alloc)
```

Comment:
if both `hasRowID` and `hasAutoIncID` are true, there will be 2 `RowIDAllocType` allocator?